obs/packages.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package obs
+
+import (
+	"bytes"
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// Package contains information about an OBS Package
+type Package struct {
+	XMLName     xml.Name `json:"directory" xml:"directory"`
```

Comment:
According to the [API docs](https://api.opensuse.org/apidocs/index#/Sources%20-%20Packages/get_source__project_name___package_name___meta), the XML tag is `packages`, not `directory`